### PR TITLE
added strformat_lab.py and mailroom.py to session03 file

### PR DESCRIPTION
**Pull request by ejnielse**

### CODE REVIEW

File: students/ejnielse/session03/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        #sum by name and find number of donations
+        for i, g in groupby(sorted(donor_list), key = lambda x: x[0]):
+            donor_list_2.append([i, sum(v[1] for v in g)])
+        donor_list_2.sort(key=lambda x:x[1], reverse = True)
+        dict_1={}
+        for i in donor_list:
+            if i[0] not in dict_1:
+                dict_1[i[0]] = 1
+            else:
+                dict_1[i[0]] += 1
```
Comment (mdgregor) [inline students/ejnielse/session03/mailroom.py]: For example, with this data structure you can simplify this as:
```
>>> donor_list = [('Daniel Cormier', [6345, 78450.67, 15450.50]),
...               ('Jon Jones', [100000.00]),
...               ('Robert Whittaker', [32675.00, 43750.67, 234.95]),
...               ('Tyron Woodley', [245000.00, 325.50]),
...               ('Khabib Nurmagomedov', [500000.00, 1000000.00])]
>>> donor_list
[('Daniel Cormier', [6345, 78450.67, 15450.5]), ('Jon Jones', [100000.0]), ('Robert Whittaker', [32675.0, 43750.67, 234.95]), ('Tyron Woodley', [245000.0, 325.5]), ('Khabib Nurmagomedov', [500000.0, 1000000.0])]
>>> 
>>> for donor in donor_list:
...     total_donations = sum(donor[1])
...     print(f"Total donations for {donor[0]} is {total_donations}")
... 
Total donations for Daniel Cormier is 100246.17
Total donations for Jon Jones is 100000.0
Total donations for Robert Whittaker is 76660.62
Total donations for Tyron Woodley is 245325.5
Total donations for Khabib Nurmagomedov is 1500000.0
>>> 

```

Comment (mdgregor) [reply students/ejnielse/session03/mailroom.py]: you can also edit the above a bit to get averages, numbers of donations etc
```
number_of_donations = len(donor[1])
average = total_donations / number_of_donations
```

Comment (mdgregor) [reply students/ejnielse/session03/mailroom.py]:
```
>>> for donor in donor_list:
...     total_donations = sum(donor[1])
...     number_of_donations = len(donor[1])
...     average = total_donations / number_of_donations
...     print(f"{donor[0]} total={total_donations} average={average} number={number_of_donations}")
... 
Daniel Cormier total=100246.17 average=33415.39 number=3
Jon Jones total=100000.0 average=100000.0 number=1
Robert Whittaker total=76660.62 average=25553.539999999997 number=3
Tyron Woodley total=245325.5 average=122662.75 number=2
Khabib Nurmagomedov total=1500000.0 average=750000.0 number=2
```

**Comment by ejnielse** [comment]
Updated with changes

**Pull request closed without merging**